Platform/Shared/VMRemovableDrivesView.swift

```diff
@@ -31,44 +31,43 @@ struct VMRemovableDrivesView: View {
     }
     
     init(vm: UTMVirtualMachine) {
-        self.vm = vm
+        self._vm = State<UTMVirtualMachine>(initialValue: vm)
         self.config = vm.configuration
         self.sessionConfig = vm.viewState
     }
     
     var body: some View {
-        Group {
+        return Group {
             if config.shareDirectoryEnabled {
                 HStack {
                     Label("Shared Directory", systemImage: "externaldrive.badge.person.crop")
                     Spacer()
                     Text(sessionConfig.sharedDirectoryPath ?? "").truncationMode(.head)
                     Button(action: clearShareDirectory, label: {
                         Text("Clear")
-                    }).disabled(vm.viewState.suspended)
+                    }).disabled(sessionConfig.suspended)
                     Button(action: { shareDirectoryFileImportPresented.toggle() }, label: {
                         Text("Browse")
-                    }).disabled(vm.viewState.suspended)
+                    }).disabled(sessionConfig.suspended)
                 }.fileImporter(isPresented: $shareDirectoryFileImportPresented, allowedContentTypes: [.folder], onCompletion: selectShareDirectory)
             }
-            ForEach(vm.drives) { drive in
-                if drive.status != .fixed {
-                    let path = sessionConfig.path(forRemovableDrive: drive.name ?? "") ?? ""
-                    HStack {
-                        Label("Interface: \(drive.interface ?? "")", systemImage: drive.imageType == .CD ? "opticaldiscdrive" : "externaldrive")
-                        Spacer()
-                        Text(path)
-                            .lineLimit(1)
-                            .truncationMode(.head)
+            ForEach(vm.drives.filter { $0.status != .fixed }) { drive in
+                HStack {
+                    DriveLabel(drive: drive)
+                    Button(action: {
```

Comment:
Are you doing "Icon - Button - Path" instead of "Icon - Path - Button" for a reason? I like the latter because it's more consistent with existing UI (path - button).

Reply:
My thinking was that I want to separate a slightly destructive action (clear) from a non-destructive action (browse). If you like it as well, I can change the shared directory UI to match. Here's how that would look (not pushed yet).
![RemovableDrives - 1](https://user-images.githubusercontent.com/12073163/101609472-4adb4c00-3a07-11eb-9760-d2b95eeef870.jpeg)
![RemovableDrives - 2](https://user-images.githubusercontent.com/12073163/101609479-4d3da600-3a07-11eb-9cec-be9590ea90b4.jpeg)
![RemovableDrives - 3](https://user-images.githubusercontent.com/12073163/101609485-4dd63c80-3a07-11eb-8d17-28a2919e921a.jpeg)
![RemovableDrives - 4](https://user-images.githubusercontent.com/12073163/101609490-4e6ed300-3a07-11eb-8a38-53e94fb08ecb.jpeg)

Reply:
That’s a good point. But is it possible to swap the two and align the clear button? Maybe that would need a geometry reader...

Reply:
Sounds good, I'll see what I can come up with!

Reply:
I have come up with an alternative design that was easier to implement, and looks very clean. Instead of the _Browse_ and _Clear_ buttons, the "Shared Directory" label on the left is now tappable and reveals a menu with _Browse_ and _Clear_ buttons. I also changed the icons for the VM details to be the same color as the text. That makes more sense anyway because they aren't interactive.

![IMG_0079](https://user-images.githubusercontent.com/12073163/101999097-c6741c00-3cd9-11eb-8038-837e5a4fdf9d.jpg)

When a VM is suspended, it becomes greyed out:

![IMG_0080](https://user-images.githubusercontent.com/12073163/101999095-c542ef00-3cd9-11eb-8a9e-2936d5f2303e.jpg)

Tapping the button makes a menu pop up just above it, containing the _Browse_/_Clear_ buttons.

![IMG_0088](https://user-images.githubusercontent.com/12073163/101999090-c2e09500-3cd9-11eb-8f3e-e0fb645d5ae3.jpg)

If the drive is ejected or no shared path is set, the _Clear_ button isn't shown.

![IMG_0087](https://user-images.githubusercontent.com/12073163/101999091-c411c200-3cd9-11eb-9efd-28bf6af0c000.jpg)

I realize that this doesn't fulfill the goal of separating the _Clear_ from the _Browse_ button, however with the Menu approach the _Browse_ button is closer to the initial touch location than the _Clear_ button so it should be fine. I tried making the _Clear_ button's text color red but that doesn't work inside a Menu.